### PR TITLE
Ignoring empty query string parameters sent to search endpoint

### PR DESCRIPTION
**Pull request by clintonb**
These empty parameters cause errors, resulting in failed responses.

ECOM-5685

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review

File: course_discovery/apps/api/filters.py [hunk 1]
```diff
@@ -59,14 +60,29 @@ def build_query(self, **query_filters):
         return query
 
 
-class HaystackFacetFilterWithQueries(HaystackFacetFilter):
+class HaystackRequestFilterMixin:
+    @staticmethod
```
Comment (schenedx) [inline course_discovery/apps/api/filters.py:64]: #Nitpick Comments for the whole mixin class would help the readability of the code a lot.

**Review by schenedx — APPROVED** [review]
👍

File: course_discovery/apps/api/filters.py (hunk 1 above)
Comment (clintonb) [reply course_discovery/apps/api/filters.py:64]: Other than something along the lines of "Mixin for Haystack requests," which is obvious from the name, I'm not sure what would make this more readable. The only method is well-commented.

**Pull request merged**